src/RestApi/StoreApi/Utilities/CartController.php

```diff
@@ -209,6 +209,24 @@ public function get_cart_coupons( $callback = null ) {
 		return $callback ? array_filter( wc()->cart->get_applied_coupons(), $callback ) : array_filter( wc()->cart->get_applied_coupons() );
 	}
 
+	/**
+	 * Selects a shipping rate.
+	 *
+	 * @param array $rate_ids Shipping rate ids.
+	 */
+	public function select_shipping_rate( $rate_ids ) {
+		$cart                    = $this->get_cart_instance();
+		$chosen_shipping_methods = WC()->session->get( 'chosen_shipping_methods' ) ? WC()->session->get( 'chosen_shipping_methods' ) : array();
+		$posted_shipping_methods = $rate_ids ? wc_clean( wp_unslash( $rate_ids ) ) : array();
```

Comment:
Doing this twice now - https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1794/files#diff-f535afef8e9431032f53589ce1ed812fR145